mhkit/wave/io/swan.py

```diff
@@ -0,0 +1,293 @@
+                            data.extend([np.nan]) 
+                    else:
+                        data.extend([float(vals)])
+            else:                
+                data.extend([float(val) for val in columns_data])             
+                
+            dataDict[variable][index_number] = data
+                
+    metaData = pd.DataFrame(metaDict).T        
+    f.close()
+    
+    for var in metaData.vars.values: 
+        df = pd.DataFrame(dataDict[var]).T        
+        varCols =  metaData[metaData.vars == var].cols.values.tolist()[0]
+        colsDict = dict(zip(df.columns.values.tolist(), varCols))
+        df.rename(columns=colsDict)
+        unitMultiplier = metaData[metaData.vars == var].unitMultiplier.values[0]
+        dataDict[var] = df * unitMultiplier 
+    
+    metaData.pop('cols')
+    metaData = metaData.set_index('vars').T.to_dict()           
+    return dataDict, metaData              
+    
+
+def _read_block_mat(output_file):
+    '''
+    Reads in SWAN matlab output and creates a dictionary of DataFrames
+    for each swan output variable.
+    
+    Parameters
+    ----------
+    output_file: str
+        filename to import
+        
+    Returns
+    -------
+    dataDict: Dictionary
+        Dictionary of DataFrame of swan output variables
+    '''
+    assert isinstance(output_file, str), 'output_file must be of type str'
+    assert isfile(output_file)==True, 'output file not found please chact name/ path'
+        
+    dataDict = loadmat(output_file, struct_as_record=False, squeeze_me=True)
+    removeKeys = ['__header__', '__version__', '__globals__']
+    for key in removeKeys:
+        dataDict.pop(key, None)
+    for key in dataDict.keys():
+        dataDict[key] = pd.DataFrame(dataDict[key])
+    return dataDict
+   
+    
+def _parse_line_metadata(line):
+    '''
+    Parses the variable metadata into a dictionary
+    
+    Parameters
+    ----------
+    line: str
+        line from block swan data to parse
+        
+    Returns
+    -------
+    metaDict: Dictionary
+        Dictionary of variable metadata
+    '''
+    assert isinstance(line, str), 'line must be of type str'
+    
+    metaDict={}        
+    meta=re.sub('\s+', " ", line.replace(',', ' ').strip('% \n').replace('**', 'vars:'))
+    mList = meta.split(':')
+    elms = [elm.split(' ') for elm in mList]
+    for elm in elms:
+        try:
+            elm.remove('')
+        except:
+            pass                
+    for i in range(len(elms)-1):
+        elm = elms[i]
+        key = elm[-1]
+        val = ' '.join(elms[i+1][:-1])
+        metaDict[key] = val
+    metaDict[key] = ' '.join(elms[-1]) 
+        
+    return metaDict    
+
+
+def dictionary_of_block_to_table(dictionary_of_DataFrames, names=None):
+    '''
+    Converts a dictionary of structured 2D grid SWAN block format 
+    x (columns),y (index) to SWAN table format x (column),y (column), 
+    values (column)  DataFrame.
+    
```

Comment:
I added a commit to sort block_to_table by xy pair. This will impact both of the functions you mention above as the dictionary of blocks is a wrapper around block_to_table.